static/sass/_snapcraft_custom.scss

```diff
@@ -0,0 +1,12 @@
+@mixin snapcraft-custom {
+  .snapcraft-hero {
+    background-image: url('#{$assets-path}19da7e00-snapcraft-hero-background.png');
+    background-position: 50% 0;
+    background-repeat: no-repeat;
+    background-size: 1440px 1440px;
+  }
+
+  .u-width-auto {
```

Comment:
This doesn't do anything where used. The buttons already have `width: auto`

Reply:
I think what @lyubomir-popov used it for is mobile buttons - to prevent them from getting full width on mobile.

Reply:
@anthonydillon wasn't aware of the strip image, thanks for pointing it out. It replicates most of what's needed here, apart from:
```background-size: {exact image dimensions; background-position: center 0;}```

Unlike ```background-size:cover```,  the above give you full control over how background image and content in the foreground overlap.

If you look at this element of the background in the strip example, it moves as you resize, and eventually goes underneath the text.
![image](https://user-images.githubusercontent.com/2741678/32548049-30dffd72-c47c-11e7-87c6-c81328bb913e.png)

![image](https://user-images.githubusercontent.com/2741678/32548020-184ac2b0-c47c-11e7-93e0-c78d22164448.png)

Specifying exact dimensions and an "anchoring" point ensures the image remains exactly as designed, which is very handy when you have predictable length text due to max-width.

You can see that when you resize the screen. The diagonal runs through the dot in the last paragraph.

Another example (from maas) is this diagonal, which I can ensure will run precisely across the play button in the foreground:
![image](https://user-images.githubusercontent.com/2741678/32548236-e51ac74a-c47c-11e7-9163-ef8fafaf3f38.png)

Reply:
@anthonydillon As to the width: auto, I was trying to prevent this from happening but I should have at least put it in a media query:
![image](https://user-images.githubusercontent.com/2741678/32548464-a5560808-c47d-11e7-9970-64dc6c42214b.png)
Full width buttons look nice on mobile (up to a certain breakpoint, and I'd argue 767px is far too wide) if you just have one. But when you have 2 as in this case, it looks too disjointed (especially when we have or in between the two).

Reply:
Add the background-size styling to the element too. It's very bespoke styling, therefore, the same as using a bespoke class.

Reply:
Regarding the `width: auto` I didnt check small screens. Thanks.

Reply:
Just to confirm about full width buttons all the way up to 767px - @anthonydillon 
![image](https://user-images.githubusercontent.com/2741678/32567749-251d367a-c4b4-11e7-9cdd-fc206558919d.png) 


Was it designed to be like this, or a missing media query between $breakpoint-small and $breakpoint-medium ?

Text + padding seems to make a lot more sense to me, especially in the context of left-aligned preceding copy:

![image](https://user-images.githubusercontent.com/2741678/32567916-acb65242-c4b4-11e7-8dc2-4731268d7532.png)

Reply:
Yes it was designed this way mainly due to a lack of a breakpoint lower then 768px. You could go full width with the buttons and hide the `or` on smaller screens?

Reply:
@anthonydillon just had a word with @kwm14 about full width buttons, as there is a compromise involved in both scenarios - the simplest solution seems to be a design decision to not have full width buttons unless we can have an upper limit on their width, at a much smaller width than 767px. 

So just dropping the ```width: 100%``` in vanilla and removing the overrides here once vanilla is updated. I'll make a vanilla proposal for the button patterns.

Reply:
@anthonydillon are we still using set breakpoints in Vanilla? That's mildly disappointing.

Reply:
@nottrobin Yes, here are the [available breakpoints](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_settings_breakpoints.scss). What would you like to see instead?